libs/cart/src/facades/cart/cart.facade.spec.ts

```diff
@@ -559,6 +569,81 @@ describe('DaffCartFacade', () => {
     });
   });
 
+  describe('paymentId$', () => {
+    let mockPayment: DaffCartPaymentMethod;
+    let cart: DaffCart;
+
+    beforeEach(() => {
+      cart = cartFactory.create();
+      mockPayment = paymentFactory.create();
+    });
+
+    describe('when the cart does not have a payment', () => {
+      beforeEach(() => {
+        store.dispatch(new DaffCartLoadSuccess({
+          ...cart,
+          payment: null
+        }));
+      });
+
+      it('should not emit', () => {
+        const expected = cold('-');
+        expect(facade.paymentId$).toBeObservable(expected);
+      });
+    });
+
+    describe('when the cart does not have a payment method', () => {
+      beforeEach(() => {
+        store.dispatch(new DaffCartLoadSuccess({
+          ...cart,
+          payment: {
+            ...mockPayment,
+            method: null
+          }
+        }));
+      });
+
+      it('should not emit', () => {
+        const expected = cold('-');
+        expect(facade.paymentId$).toBeObservable(expected);
+      });
+    });
+
+    describe('when the cart\'s payment method is not defined in the map', () => {
+      beforeEach(() => {
+        store.dispatch(new DaffCartLoadSuccess({
+          ...cart,
+          payment: {
+            ...mockPayment,
+            method: 'not in the map'
+          }
+        }));
+      });
+
+      it('should return undefined', () => {
+        const expected = cold('a', {a: undefined});
+        expect(facade.paymentId$).toBeObservable(expected);
+      });
+    });
+
+    describe('when the cart\'s payment method is defined in the map', () => {
+      beforeEach(() => {
+        store.dispatch(new DaffCartLoadSuccess({
+          ...cart,
+          payment: {
+            ...mockPayment,
+            method: paymentMethod
+          }
+        }));
+      });
+
+      it('should return the mapped payment ID', () => {
```

Comment:
I think this should be "should return the platform agnostic payment ID"

Reply:
changed